pkg/dogstatsd/parse_events.go

```diff
@@ -84,7 +86,7 @@ func parseHeader(rawHeader []byte) (eventHeader, error) {
 	rawTextLength := rawLengths[sepIndex+1:]
 
 	// Convert title length to workable type and do a basic validity check on value
-	titleLength, err := parseInt64(rawTitleLength)
+	titleLength, err := strconv.Atoi(*(*string)(unsafe.Pointer(&rawTitleLength)))
```

Comment:
This feels really strange, why is it inlining the function (thus exposing the horrible hack at every call point) ? And why is it using `Atoi` instead of `ParseInt` when most of the other changes are actually `Atoi -> ParseInt/ParseUint`

Reply:
I suspect it's because it's a `[]byte` and this is trying to do a "free" conversion to string?  But yeah, odd that it's reversing the tendency elsewhere!

Reply:
yeah, i copied the "free" string conversion design pattern because I figured it was implemented for a reason

I think CodeQL pinged because of the `int` cast below (https://github.com/stephengroat/datadog-agent/pull/1/files/fb14adbc126746442c73b968d30dfb3caa3694ae#diff-bc88e0c1fc63c2348a0857c8fcdc7c8f099100129267858323d19649eb6825c9L104-L105) and it wants to keep consistency. on a 32 bit system, there could be a potential loss of precision when moving from `int64` to `int` i guess?

Reply:
Alert from CodeQL for reference:
https://github.com/stephengroat/datadog-agent/security/code-scanning/48
https://github.com/stephengroat/datadog-agent/security/code-scanning/49?